app/src/test/java/org/simple/clinic/appconfig/AppConfigRepositoryTest.kt

```diff
@@ -30,4 +36,83 @@ class AppConfigRepositoryTest {
         .assertValue(FetchSucceeded(countries))
         .assertNoErrors()
   }
+
+  @Test
+  fun `failed network calls to fetch the app manifest to return an appropriate failure result`() {
+    // given
+    val cause = ConnectException()
+    whenever(api.fetchManifest()).thenReturn(Single.error(cause))
+
+    // then
+    repository
+        .fetchAppManifest()
+        .test()
+        .assertValue(FetchError(ResolvedError.NetworkRelated(cause)))
+        .assertNoErrors()
+  }
+
+  @Test
+  fun `server errors when fetching the app manifest should return an appropriate failure result`() {
+    // given
+    val cause = httpException(500)
+    whenever(api.fetchManifest()).thenReturn(Single.error(cause))
+
+    // then
+    repository
+        .fetchAppManifest()
+        .test()
+        .assertValue(FetchError(ResolvedError.ServerError(cause)))
+        .assertNoErrors()
+  }
+
+  @Test
+  fun `unauthenticated errors when fetching the app manifest should return an appropriate failure result`() {
```

Comment:
Rename to `401 errors when fetching the app manifest should return failure result as unauthenticated`